RecoTauTag/HLTProducers/src/TauTagFilter.cc

```
@@ -0,0 +1,156 @@
+    desc.add<std::vector<int>>("seedTypes",
+                               {trigger::TriggerL1Tau, trigger::TriggerL1Jet, trigger::TriggerTau, trigger::TriggerJet})
+        ->setComment("list of seed object types");
+    desc.add<std::string>("selection", "0")->setComment("selection formula");
+    desc.add<double>("minPt", 20)->setComment("minimal tau pt");
+    desc.add<double>("maxEta", 2.5)->setComment("maximal tau abs(eta)");
+    desc.add<bool>("usePtCorr", false)->setComment("use multiplicative tau pt corrections");
+    desc.add<bool>("matchWithSeeds", false)->setComment("apply match with seeds");
+    desc.add<double>("matchingdR", 0.5)->setComment("deltaR for matching with seeds");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& eventsetup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override {
+    if (saveTags())
+      filterproduct.addCollectionTag(tausSrc_);
+
+    int nTauPassed = 0;
+
+    const auto tausHandle = event.getHandle(tausToken_);
+    const auto& taus = *tausHandle;
+
+    std::vector<LorentzVectorM> seed_p4s;
+    if (matchWithSeeds_) {
+      const auto& seeds = event.get(seedsSrc_);
+      for (const int seedType : seedTypes_) {
+        if (seedType == trigger::TriggerL1Tau) {
+          extractMomenta<l1t::TauVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerL1Jet) {
+          extractMomenta<l1t::JetVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerTau) {
+          extractMomenta<std::vector<reco::PFTauRef>>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerJet) {
+          extractMomenta<std::vector<reco::PFJetRef>>(seeds, seedType, seed_p4s);
+        } else
+          throw cms::Exception("Invalid seed type", "PNetTauTagFilter::hltFilter") << "Invalid seed type: " << seedType;
```

Is not better to throw an exception already at C-tor if there is a seed type which is not supported?